### PR TITLE
Add integration test for upgrading from edge

### LINKED ISSUE #4471: Can't upgrade Linkerd from edge 20.5.3 to 20.5.4 due to render error
### What is the issue?
When upgrading from 20.5.3 to 20.5.4, I am getting the following error after running `linkerd upgrade`:
```
× Could not render upgrade configuration: render error in "grafana/templates/grafana.yaml": template: grafana/templates/grafana.yaml:124:24: executing "grafana/templates/grafana.yaml" at <.Values.image.name>: can't evaluate field name in type interface {}
For troubleshooting help, visit: https://linkerd.io/upgrade/#troubleshooting
```

### How can it be reproduced?
```
curl https://run.linkerd.io/install-edge | sh -
linkerd upgrade --context $CLUSTER --ha | kubectl --context $CLUSTER apply -f -
```

### Logs, error output, etc

See above

#### `linkerd check` output

```text
$ linkerd check
kubernetes-api
--------------
√ can initialize the client
√ can query the Kubernetes API

kubernetes-version
------------------
√ is running the minimum Kubernetes API version
√ is running the minimum kubectl version

linkerd-existence
-----------------
√ 'linkerd-config' config map exists
√ heartbeat ServiceAccount exist
√ control plane replica sets are ready
√ no unschedulable pods
√ controller pod is running
√ can initialize the client
√ can query the control plane API

linkerd-config
--------------
√ control plane Namespace exists
√ control plane ClusterRoles exist
√ control plane ClusterRoleBindings exist
√ control plane ServiceAccounts exist
√ control plane CustomResourceDefinitions exist
√ control plane MutatingWebhookConfigurations exist
√ control plane ValidatingWebhookConfigurations exist
√ control plane PodSecurityPolicies exist

linkerd-identity
----------------
√ certificate config is valid
√ trust anchors are using supported crypto algorithm
√ trust anchors are within their validity period
√ trust anchors are valid for at least 60 days
√ issuer cert is using supported crypto algorithm
√ issuer cert is within its validity period
√ issuer cert is valid for at least 60 days
√ issuer cert is issued by the trust anchor

linkerd-api
-----------
√ control plane pods are ready
√ control plane self-check
√ [kubernetes] control plane can talk to Kubernetes
√ [prometheus] control plane can talk to Prometheus
√ tap api service is running

linkerd-version
---------------
√ can determine the latest version
√ cli is up-to-date

control-plane-version
---------------------
‼ control plane is up-to-date
    is running version 20.5.3 but the latest edge version is 20.5.4
    see https://linkerd.io/checks/#l5d-version-control for hints
‼ control plane and cli versions match
    control plane running edge-20.5.3 but cli running edge-20.5.4
    see https://linkerd.io/checks/#l5d-version-control for hints

linkerd-ha-checks
-----------------
√ pod injection disabled on kube-system
√ multiple replicas of control plane pods

linkerd-addons
--------------
√ linkerd-config-addons config map exists

linkerd-grafana
---------------
√ grafana add-on service account exists
√ grafana add-on config map exists
√ grafana pod is running

Status check results are √
```

### Environment

- Kubernetes Version: v1.17.0
- Cluster Environment: (GKE, AKS, kops, ...) AKS
- Host OS: Mac
- Linkerd version: 20.5.3

### Possible solution
Looks like it targets this line in `grafana.yaml`: `       image: {{.Values.grafanaImage}}:{{default .Values.global.linkerdVersion .Values.controllerImageVersion}}`

### PR DESCRIPTION
**Pull request by kleimkuhler**
This adds an integration test for upgrading from the latest edge to the current
build.

Closes #4471 

Signed-off-by: Kevin Leimkuhler <kevin@kleimkuhler.com>

### CODE REVIEW

File: bin/test-run [hunk 1]
```diff
@@ -1,7 +1,8 @@
 #!/bin/bash
 
 # This script executes the full suite of integration tests, in series:
-# 1. upgrade_integration_tests
+# 1. upgrade_stable_integration_tests
+# 1. upgrade_edge_integration_tests
```
Comment (alpeb) [inline bin/test-run:5]: Nit: numbering

File: test/install_test.go [hunk 2]
```diff
@@ -235,7 +235,7 @@ func TestInstallOrUpgradeCli(t *testing.T) {
 		}
 
 		// prepare for stage 2
-		args = append([]string{"control-plane"}, args...)
+		args = append([]string{"control-plane", "--addon-overwrite"}, args...)
```
Comment (alpeb) [inline test/install_test.go:238]: I think we shouldn't add this flag as it doesn't reflect the usual upgrade scenario. Without it the upgrade fails from edge-20.5.3 like described in #4471, but it's green without it using the current latest edge.

Comment (kleimkuhler) [reply test/install_test.go:238]: I don't think this works for any edge right now. I ran into this issue when upgrading from `edge-20.6.1` as well. I originally opened an issue for it, but closed it (#4556).

Comment (alpeb) [reply test/install_test.go:238]: Ah sorry, I was testing in a different branch... I confirmed it's failing upgrading from Friday's edge as well. From what I can see, the resulting upgrades with and without `--from-manifests` are different. The former isn't touching Grafana's version while the latter is. @Pothulapati can you lend a hand to see what's going on?

Comment (kleimkuhler) [reply test/install_test.go]: This was updated to get `configmaps/linkerd-config-addons` if it exists.

**Comment by alpeb** [comment]
This might depend on #4571 in order to pass, so I'm merging that one now.

File: test/install_test.go [hunk 3]
```diff
@@ -247,20 +247,29 @@ func TestInstallOrUpgradeCli(t *testing.T) {
...
+		resources = append(resources, "configmaps/"+k8s.ConfigConfigMapName, "secrets/"+k8s.IdentityIssuerSecretName)
+
+		// If `linkerd-config-addons` exists, add it to the resources to get
+		_, err := TestHelper.Kubectl("", "--namespace", TestHelper.GetLinkerdNamespace(), "get", "configmaps/"+k8s.AddOnsConfigMapName)
+		if err != nil {
+			resources = append(resources, "configmaps/"+k8s.AddOnsConfigMapName)
+		}
+
+		args := append([]string{"--namespace", TestHelper.GetLinkerdNamespace(), "get"}, resources...)
+		args = append(args, "-o yaml")
```
Comment (alpeb) [inline test/install_test.go:260]: Maybe the test is failing also because `-o yaml` is used in a single arg?

Comment (kleimkuhler) [reply test/install_test.go:260]: Yes definitely it. Thanks!

**Comment by grampelberg** [comment]
@kleimkuhler what do we need to do to get this merged?

**Comment by kleimkuhler** [comment]
@grampelberg It needs to be admin merged and then the expected checks need to be updated in GitHub settings.

This PR changes a check name from `Integration tests (upgrade)` to `Integration test (upgrade_stable)`. All checks have passed, it just expects a test that is changed by this PR.

**Comment by grampelberg** [comment]
@admc can you help out?
